metricflow-semantics/metricflow_semantics/collection_helpers/dedupe.py

```diff
@@ -0,0 +1,15 @@
+from __future__ import annotations
+
+from typing import Dict, Iterable, Tuple, TypeVar
+
+IterableT = TypeVar("IterableT")
+
+
+def ordered_dedupe(*iterables: Iterable[IterableT]) -> Tuple[IterableT, ...]:
+    """De-duplicates the items in the iterables while preserving the order."""
+    ordered_results: Dict[IterableT, None] = {}
+    for iterable in iterables:
+        for item in iterable:
+            ordered_results[item] = None
+
+    return tuple(ordered_results.keys())
```

Comment:
This is a less robust custom implementation of the (admittedly poorly named) unique_everseen function in more-itertools.

Since dbt-semantic-interfaces depends on more-itertools we should add the dependency with the same version range and use it here as well.

Reply:
Actually I just noticed this takes `*iterables`, so it's a less robust implementation of `unique_everseen(itertools.chain(iterables))`. The point still stands, we should use the libraries, especially where iterator collection helpers are concerned.

Reply:
Sure, we can make that change, but let me check about the "minimize dependencies" request and check how that method works with types.

Reply:
I already looked at the dependencies, that's not an issue. The typing is another matter, more-itertools is an old library and these methods date back a long way.